tests/integration/strategy_tests.cpp

```diff
@@ -599,3 +614,15 @@ TEST(NumaPolicyTest, Location) {
 }
 
 #endif // defined(UMPIRE_ENABLE_NUMA)
+
+
+int main( int argc, char* argv[] )
+{
+  int result = 0;
+  testing::InitGoogleTest( &argc, argv );
+  result = RUN_ALL_TESTS();
+
+  umpire::ResourceManager::getInstance().finalize();
```

Comment:
This is kind of annoying to do for every test.

Reply:
You only have to do it for tests that you want to run memcheck on. If we were to change
``` c++
ResourceManager::getInstance()
{
  if (!s_resource_manager_instance) {
    s_resource_manager_instance = new ResourceManager();
  }

  UMPIRE_LOG(Debug, "() returning " << s_resource_manager_instance);
  return *s_resource_manager_instance;
}
```
 to
``` c++
ResourceManager::getInstance()
{
  static ResourceManager rm;
  s_resource_manager_instance = &rm;
  return rm;
}
```
and then deleted the `ResourceManager` copy/move constructors and the copy/move assignment operators then we could use `~ResourceManager()` to do this. However you already had `finalize` methods elsewhere in the code and I wanted to limit my changes.

Reply:
I don't think we need to fix this, instead we can suppress it.

Reply:
Was your comment intended for the conversation below (The tests case I commented out)? If not I'm not sure I understand what you mean by suppressing it.

Reply:
I meant, don't have the custom main for the tests, and just suppress the warnings they generate from not calling `ResourceManager::finalize()`

Reply:
Although the approach you suggest using `static ResourceMangager rm` might be a good solution. Do you know of any major pros/cons of these two different approaches to implement a singleton?

Reply:
It's thread safe (unlike the current implementation although you could just put a lock around the check). It doesn't involve an `if` check every time and the destructor will automatically be called. The one downside is that with a pointer `ResourceManager::getInstance()' should be valid throughout program termination so any other static objects can access it in their destructor. However if we use `static ResourceMangager rm` then it would depend on the initialization order. @rrsettgast knows more about this than I do. 

https://stackoverflow.com/questions/1008019/c-singleton-design-pattern/1008289#1008289
https://stackoverflow.com/questions/1661529/is-meyers-implementation-of-the-singleton-pattern-thread-safe

Reply:
Do you have other static objects that require access to the ResourceManager? If not, use the Myers singleton. Here is another discussion.

https://stackoverflow.com/questions/13047526/difference-between-singleton-implemention-using-pointer-and-using-static-object